# Nhóm 21
**Trần Nhật Huy - 1612272**

**Nguyễn Duy Hưng - 1512222**

# Nguồn gốc dữ liệu
Đây là dữ liệu được lấy từ API [Dark Sky](https://darksky.net/dev)

Dữ liệu là thông tin thời tiết từ `01-01-2010` đến `31/12/2011` ở `Hồ Chí Minh Việt Nam`

# Data Understanding

In [4]:
df = pd.read_csv("weather.csv")
df

time  temperature  apparentTemperature  dewPoint  humidity  \
0    1262304000        73.45                75.07     72.69      0.97   
1    1262390400        71.67                72.95     69.81      0.94   
2    1262476800        75.25                76.89     73.39      0.94   
3    1262563200        75.25                76.64     71.61      0.88   
4    1262649600        75.25                76.89     73.39      0.94   
..          ...          ...                  ...       ...       ...   
725  1324944000        71.67                72.95     69.81      0.94   
726  1325030400        73.45                75.18     73.39      1.00   
727  1325116800        71.67                72.44     66.21      0.83   
728  1325203200        73.45                74.65     69.81      0.88   
729  1325289600        73.45                74.91     71.61      0.94   

     pressure  windSpeed  windBearing  cloudCover  uvIndex  visibility  
0      1010.3       6.72         71.0        0.50      0.0         NaN  
1      1011.0       2.29         41.0        0.44      0.0       4.974  
2      1011.6       1.16         30.0        0.75      0.0       4.346  
3      1010.9       3.46         71.0        0.75      0.0       6.216  
4      1010.1       6.91        120.0        0.44      0.0       6.216  
..        ...        ...          ...         ...      ...         ...  
725    1012.0       4.61          0.0        0.75      0.0       5.591  
726    1013.7       4.61         11.0        0.75      0.0       3.108  
727    1012.8       2.29         11.0        0.44      0.0       6.216  
728    1012.4       4.61          0.0        0.44      0.0       6.216  
729    1013.1       4.61         11.0        0.75      0.0       6.216  

[730 rows x 11 columns]

Dữ liệu gồm có **730 dòng** và **11 cột** 

Ý nghĩa của các cột:
- **`time`**: ngày
- **`temperature`**: nhiệt độ
- **`apparentTemperature`**: nhiệt độ biểu kiến, nhiệt độ con người cảm nhận được
- **`dewPoint`**: điểm sương, là nhiệt độ mà ở đó thành phần hơi nước trong khối không khí ngưng đọng thành nước lỏng
- **`humidity`**: độ ẩm
- **`pressure`**: áp suất
- **`windSpeed`**: tốc độ gió
- **`windBearing`**: hướng gió được tính bắt đầu từ hướng Bắc (0 độ) và tính theo chiều kim đồng hồ. Nếu tốc độ gió là 0 thì giá trị này sẽ không có
- **`cloudCover`**: độc che phủ của mây
- **`uvIndex`**: chỉ số tử ngoại, là một chỉ số đo lường theo tiêu chuẩn quốc tế về cường độ của bức xạ tử ngoại từ mặt trời tại một địa điểm cụ thể vào một ngày cụ thể
- **`visibility`**: tầm nhìn

In [5]:
df.describe()

time  temperature  apparentTemperature    dewPoint    humidity  \
count  7.300000e+02   730.000000           730.000000  730.000000  730.000000   
mean   1.293797e+09    77.140397            80.739726   74.183274    0.911233   
std    1.821978e+07     3.639286             6.316240    3.891820    0.083330   
min    1.262304e+09    66.270000            66.480000   60.800000    0.610000   
25%    1.278050e+09    75.250000            76.640000   72.865000    0.840000   
50%    1.293797e+09    77.060000            79.150000   75.190000    0.940000   
75%    1.309543e+09    78.880000            83.380000   76.990000    1.000000   
max    1.325290e+09    86.060000            99.510000   80.580000    1.000000   

          pressure   windSpeed  windBearing  cloudCover     uvIndex  \
count   723.000000  730.000000   662.000000  726.000000  726.000000   
mean   1009.774412    4.726534   143.509063    0.491928    0.643251   
std       1.970895    2.851606   109.787909    0.201766    0.479370   
min    1003.700000    0.000000     0.000000    0.190000    0.000000   
25%    1008.500000    2.290000    50.000000    0.440000    0.000000   
50%    1009.600000    4.610000   110.000000    0.440000    1.000000   
75%    1010.800000    6.720000   230.000000    0.750000    1.000000   
max    1016.800000   22.380000   350.000000    1.000000    1.000000   

       visibility  
count  715.000000  
mean     5.069698  
std      1.631170  
min      0.434000  
25%      3.730000  
50%      6.216000  
75%      6.216000  
max     10.000000

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 730 entries, 0 to 729
Data columns (total 11 columns):
time                   730 non-null int64
temperature            730 non-null float64
apparentTemperature    730 non-null float64
dewPoint               730 non-null float64
humidity               730 non-null float64
pressure               723 non-null float64
windSpeed              730 non-null float64
windBearing            662 non-null float64
cloudCover             726 non-null float64
uvIndex                726 non-null float64
visibility             715 non-null float64
dtypes: float64(10), int64(1)
memory usage: 62.9 KB


Ta thấy dữ liệu có các cột chứa giá trị thiếu bao gồm ***pressure***, ***windBearing***, ***cloudCover***, ***uvIndex***, ***visibility*** 

# Câu hỏi mà nhóm đặt ra

Nhóm muốn dựa vào data thu thập được để dự đoán thuộc tính **humidity (độ ẩm)** của **ngày hôm sau** dựa vào dữ liệu của ngày hôm trước. Việc dự đoán này sẽ giúp ích cho một số công việc như:
- Dự báo thời tiết tại một địa điểm cụ thể
- Đề phòng cho người dân các loại bệnh có thể xảy ra khi thời tiết thay đổi
- ...

Đây là câu hỏi mà nhóm tự đặt ra